src/apiData/apiCalls.js

```diff
@@ -1,16 +1,27 @@
+import React from 'react'
+
 const baseURL = 'https://rancid-tomatillos.herokuapp.com/api/v2/'
 const apiCalls = {
   fetchApiData: async (query) => {
-    const response = await fetch(`${baseURL}${query}`)
-    return await apiCalls.checkForError(response)
+    return await fetch(`${baseURL}${query}`)
   },
 
-  checkForError: response => {
-    if (!response.ok) {
-      throw new Error('Network response was not ok')
-    } 
-    
-    return response.json()
+  checkForError: status => {
+    let errorMsg;
+
+    switch (status) {
+      case 404:
+        errorMsg = 'Sorry! We can\'t find the page you\'re looking for...';
+        break;
+      case 500:
+        errorMsg = 'Internal Server Error. Our whole team are now aware.';
+        break;
+      default:
+        errorMsg = 'Oops! Request failed. Please try again.';
+    }
```

Comment:
Nice job with dynamically changing the message dependent on the status received.